tests/integration_tests/functional/test_api.py

```diff
@@ -743,3 +744,50 @@ def _drive_patch(test_microvm):
         path_on_host=test_microvm.create_jailed_resource(fs.path)
     )
     assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+
+def test_api_vsock(test_microvm_with_api):
+    """Test vsock related API commands."""
+    if test_microvm_with_api.build_feature != 'vsock':
+        pytest.skip("This test is meant only for vsock builds")
+
+    test_microvm = test_microvm_with_api
+    test_microvm.spawn()
+    test_microvm.basic_config()
+
+    response = test_microvm.vsock.put(
+        vsock_id='vsock1',
+        guest_cid=15
+    )
+    assert test_microvm.api_session.is_good_response(response.status_code)
```

Comment:
We switched from asserting on is_good_response to the actual returned status code. Please use `is_status_no_content` or whatever is the appropriate method here.